front/lib/resources/data_source_resource.ts

```diff
@@ -174,6 +138,21 @@ export class DataSourceResource extends ResourceWithVault<DataSourceModel> {
     });
   }
 
+  static async fetchById(
+    auth: Authenticator,
+    id: string,
+    options?: Omit<FetchDataSourceOptions, "limit" | "order">
+  ): Promise<DataSourceResource | null> {
+    const [dataSource] = await DataSourceResource.fetchByIds(
```

Comment:
Q: Why do we prefer doing a where in array of 1 instead of a findOne? 🙏🏻

Reply:
That's what we do everywhere

Reply:
OK but do you know why? 🙏🏻

Reply:
One code path to baseFetchWithAuthorization instead of many I believe is the rationale